src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,67 @@
+package duke.task;
+
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+import duke.exceptions.DukeException;
+
+
+/**
+ * Represents a list that stores all tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Overloaded constructor that sets the user's task list as the task list from the database.
+     *
+     * @param taskList the list received from the database
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    public void addTask(Task task) {
+        this.taskList.add(task);
+    }
+
+    /**
+     * Method to delete the task at the specified index.
+     *
+     * @param index is the index of the task to be deleted.
+     * @return the task object that is being deleted.
+     * @throws DukeException if the index specified by the user is invalid.
+     */
+    public Task deleteTask(int index) throws DukeException {
+        if (isIndexInRange(index)) {
+            return this.taskList.remove(index - 1);
+        }
+
+        throw new DukeException("Oh no! Task number does not exist in task list.");
+    }
+
+    public int size() {
```

Comment:
Maybe you can add javadoc comments for the public methods here?

Reply:
Ahh yes somehow checkstyle did not flag me out for this. Might be because its just a one liner 💭